block.go

```diff
@@ -455,21 +461,29 @@ func (r blockIndexReader) LabelValues(names ...string) (index.StringTuples, erro
 	return st, errors.Wrapf(err, "block: %s", r.b.Meta().ULID)
 }
 
-func (r blockIndexReader) Postings(name, value string) (index.Postings, error) {
-	p, err := r.ir.Postings(name, value)
-	return p, errors.Wrapf(err, "block: %s", r.b.Meta().ULID)
+func (r blockIndexReader) Postings(name, value string, reusePos index.Postings) (index.Postings, error) {
+	p, err := r.ir.Postings(name, value, reusePos)
+	if err != nil {
+		// Checking for error before wrapping saves some allocs.
```

Comment:
How does this save allocs? errors.Wrapf returns immediately if err is nil.

Reply:
The extra function call results in few allocs. Coupled that with 100s of thousands of calls, the savings in allocs is pretty significant (Same for `Series` function below).

I got that insight from the profile (`-alloc_objects` flag), and % allocs saved was significant in the benchmark.

Reply:
Function calls don't cause allocs.

Reply:
Yes, function calls don't. But some part in it apparent caused allocs. I will post the screenshots of profiles here in some time.

Reply:
Before = direct Wrapf, Now = checking error before.

## `Postings(...)`
**Before:**
![Screenshot from 2019-06-20 15-14-34](https://user-images.githubusercontent.com/15064823/59839642-9232d000-936e-11e9-8e90-6c9d682ed5c8.png)
![Screenshot from 2019-06-20 15-14-56](https://user-images.githubusercontent.com/15064823/59839662-9b23a180-936e-11e9-9c60-e484feb78897.png)

**Now:**
Can see that `ir.Postings(...)` causes no allocs. Could not get profile of `Postings(...)` in block.go as there were no allocs.
![Screenshot from 2019-06-20 15-11-29](https://user-images.githubusercontent.com/15064823/59839691-a5de3680-936e-11e9-9353-07008b946520.png)

## `Series(...)`
**Before:**
![Screenshot from 2019-06-20 15-14-41](https://user-images.githubusercontent.com/15064823/59839814-dd4ce300-936e-11e9-9d10-3f4b630c19c5.png)
(see `set.Next()`)
![Screenshot from 2019-06-20 15-15-02](https://user-images.githubusercontent.com/15064823/59839834-e76ee180-936e-11e9-9f1d-5bf275d672eb.png)

**Now:**
Could not get profile of `Series(...)` in block.go as there were no allocs.
![Screenshot from 2019-06-20 15-12-21](https://user-images.githubusercontent.com/15064823/59839855-f190e000-936e-11e9-8695-e576ba0a5d35.png)

Reply:
Wow it's quite interesting, but makes sense (:

Reply:
Also it's more clear / explcit so :+1: 

I would delete the comment though. I think it is  general knowledge that should be ensure by some linter and documented. Otherwise you need to put this comment on every `Wrapf` with reminder to avoid `Wrapf` on nil error. That does not scale well (:

Reply:
As these 2 functions are called a lot of time, it should be fine changing only at these 2 places. Also keeping this comment here would prevent it from reverting back in future as it looks natural to revert back.